src/Fixer/Basic/BracesPositionFixer.php

```diff
@@ -251,7 +251,7 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
 
                 $positionOption = 'control_structures_opening_brace';
             } elseif ($token->isGivenKind(T_VARIABLE)) {
-                $openBraceIndex = $tokens->getNextTokenOfKind($index, ['{', ';', [CT::T_PROPERTY_HOOK_BRACE_OPEN], [T_CLOSE_TAG]]);
+                $openBraceIndex = $tokens->getNextTokenOfKind($index, ['{', ';', [CT::T_CURLY_CLOSE], [CT::T_PROPERTY_HOOK_BRACE_OPEN], [T_CLOSE_TAG]]);
 
```

Comment:
```suggestion
                $nextToken = $tokens->getNextMeaningfulToken($index);
                if (!$tokens[$nextToken]->equalsAny(['=', [CT::T_PROPERTY_HOOK_BRACE_OPEN]])) {
                    continue;
                }

                $openBraceIndex = $tokens->getNextTokenOfKind($x, ['{', ';', [CT::T_PROPERTY_HOOK_BRACE_OPEN], [T_CLOSE_TAG]]);
```
I was thinking about something like this:
- first - quick "deal breaker" for having a property hook after the variable
- second - proper check if the property hook is there

The `$x` could be either `$index` (for simplicity) or `$nextToken - 1` (for microoptimization).

Reply:
not sure if I grasp your idea here, want to sent a PR with it ?

Reply:
If the variable is not followed by `=` or `T_PROPERTY_HOOK_BRACE_OPEN`, it cannot be a property with hooks.
In the given test cases it would be a fast fail for all variables inside the string.
Only for the first `$foo =  ...` variable it would still traverse through all tokens in the long statement to the statement end (`;`).